hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/snapshot/TestSnapshotUtils.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.om.snapshot;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import org.apache.hadoop.hdds.client.BlockID;
+import org.apache.hadoop.ozone.OzoneConsts;
+import org.apache.hadoop.ozone.om.helpers.OmKeyInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyLocationInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyLocationInfoGroup;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Unit test for SnapshotUtils.
+ */
+public class TestSnapshotUtils {
+  private static final long CONTAINER_ID = 1L;
+
+  private OmKeyLocationInfo createLocation(long blockId, long length) {
+    BlockID blockID = new BlockID(CONTAINER_ID, blockId);
+    // Only blockId is used in .hasSameBlockAs for the corner cases we're testing.
+    // You can expand this as needed for extra fields.
+    return new OmKeyLocationInfo.Builder()
+        .setBlockID(blockID)
+        .setLength(length)
+        .build();
+  }
+
+  private OmKeyLocationInfoGroup createLocationGroup(OmKeyLocationInfo... locs) {
+    OmKeyLocationInfoGroup group = new OmKeyLocationInfoGroup(0, Arrays.asList(locs));
+    return group;
+  }
+
+  private OmKeyInfo createOmKeyInfo(boolean hsync, OmKeyLocationInfoGroup group) {
+    return createOmKeyInfo(hsync, group, 0);
+  }
+
+  private OmKeyInfo createOmKeyInfo(boolean hsync, OmKeyLocationInfoGroup group, long objectId) {
+    return createOmKeyInfo(hsync, Collections.singletonList(group), objectId);
+  }
+
+  private OmKeyInfo createOmKeyInfo(boolean hsync, List<OmKeyLocationInfoGroup> group, long objectId) {
+    OmKeyInfo keyInfo = new OmKeyInfo.Builder()
```

Comment:
nit: I was just wondering if you can have a similar function in https://github.com/apache/ozone/blob/5c5db8e9bc5b8dcf5d586e10aa648ac838700d49/hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/request/OMRequestTestUtils.java#L559-L560 and call it here. Most of function parameters are the same.

Reply:
I'll look into this later as a separate task. We have a lot of places that do similar work, so might as well to address that holistically.